src/utils/notification-common.ts

```diff
@@ -0,0 +1,128 @@
+import { cva } from "class-variance-authority";
+
+/** The duration of the animation in the appearance and disappearance of the toast (in msec). */
+export const TOAST_ANIMATION_DURATION = 200;
+
+export function formatDate(date: Date): string {
+  const result = date.toLocaleDateString(undefined, {
+    dateStyle: "medium",
+  });
+
+  if (result === "Invalid Date") {
+    if (import.meta.env.DEV) {
+      console.warn(`Invalid 'timestamp' property: Invalid date.`);
+    }
+
+    return "";
+  }
+
+  return result;
+}
```

Comment:
Using `Number.isNaN(date.getTime()` will be more precise.

```suggestion
export function formatDate(date: Date): string | null {
  if (Number.isNaN(date.getTime()) {
    if (import.meta.env.DEV) {
      console.warn("formatDate: Invalid Date object provided");
    }
    
    return null;
  }

  return date.toLocaleDateString(undefined, {
    dateStyle: "medium",
  });
}
```